### ISSUE TITLE
How use/settings http_basic_auth? 

### ISSUE THREAD

**Issue by araslanov-e** [jeremyevans/rodauth#41]
I not understand how use http_basic_auth
```
rodauth do
  enable :http_basic_auth
end
roda do |r|
  r.rodauth
  rodauth.require_authentication
  r.root{view :content=>(rodauth.logged_in? ? "Logged In" : 'Not Logged')}
end
```
It's redirect to /login

**Comment by jeremyevans** [comment]
http_basic_auth allows users to authenticate using HTTP basic authentication, but doesn't force itself as the sole means by which to authenticate.  It's not going to return a 401 unless an Authorization request header is provided and the login/password do not match.  You could remove the `rodauth.require_authentication` line, and you'll see Logged In or Not Logged depending on whether the account is logged in.

I think it would be a good idea to add a `require_http_basic_auth` configuration method to the http_basic_auth feature that makes `require_authentication` return 401 if the request has not been authenticated.  If that's something you would like to work on, please let me know, otherwise I'll try to work on it before the next release.

**Comment by araslanov-e** [comment]
Now i used https://github.com/badosu/roda-basic-auth/, exactly what is needed
But its not supporting

**Comment by jeremyevans** [comment]
Not sure why GitHub didn't show the related commit when this was closed, but here it is:  302c53854bd9bcddfdedd316b13fba552fe26d24

**Comment by araslanov-e** [comment]
It's good. Thanks
